### PR TITLE
feat(ui): Add sort options to user list

### LINKED ISSUE #767: Sort user list by alias
**Is your feature request related to a problem? Please describe.**
Not related to a problem, more of a QoL addition.

**Describe the solution you'd like**
A capability to sort the user list by all/some column fields

**Describe alternatives you've considered**
Current UI doesn't allow for other alternatives as there are no UI elements for this specific request.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by TheCatLady**
#### Description

Adds options to sort user list by creation date / ID (default sort and current behavior), last modified date, request count, and username.

#### Screenshot (if UI related)

**Desktop view:**
image

**Mobile view:**
image

#### Todos

- [x] Successful build
- [x] Translation keys

#### Issues Fixed or Closed by this PR

- Fixes #767

### CODE REVIEW

File: server/routes/user.ts [hunk 1]
```diff
@@ -11,10 +11,43 @@ import { UserType } from '../constants/user';
 
 const router = Router();
 
-router.get('/', async (_req, res) => {
+router.get('/', async (req, res) => {
   const userRepository = getRepository(User);
 
-  const users = await userRepository.find();
+  let sortFilter: FindOneOptions<MediaRequest>['order'] = {
```
Comment (sct) [inline server/routes/user.ts:17]: Why the type `FindOneOptions` instead of `FindOptions`?

File: server/routes/user.ts [hunk 2]
```diff
@@ -11,10 +11,43 @@ import { UserType } from '../constants/user';
...
+
+  let users = await userRepository.find({
+    order: sortFilter,
+  });
+
+  switch (req.query.sort) {
+    case 'requests':
+      users = users.sort(
+        (user1, user2) => user2.requestCount - user1.requestCount
+      );
```
Comment (sct) [inline server/routes/user.ts:37]: This query will have pagination in the future. This won't work. We need to build the query to do this.

File: server/routes/user.ts [hunk 3]
```diff
@@ -11,10 +11,43 @@ import { UserType } from '../constants/user';
...
+    case 'username':
+      users = users.sort((user1, user2) => {
+        if (user1.displayName > user2.displayName) {
+          return 1;
+        }
+        if (user1.displayName < user2.displayName) {
+          return -1;
+        }
+        return 0;
+      });
```
Comment (sct) [inline server/routes/user.ts:48]: Same thing here.

File: src/components/UserList/index.tsx [hunk 4]
```diff
@@ -368,24 +377,63 @@ const UserList: React.FC = () => {
...
+            </span>
+            <select
+              id="sort"
+              name="sort"
+              onChange={(e) => {
+                setCurrentSort(e.target.value as Sort);
+              }}
+              onBlur={(e) => {
+                setCurrentSort(e.target.value as Sort);
+              }}
```
Comment (sct) [inline src/components/UserList/index.tsx:419]: We don't need an onBlur here anymore. The eslint rule for this was disabled.

File: src/components/UserList/index.tsx [hunk 5]
```diff
@@ -62,13 +62,22 @@ const messages = defineMessages({
     'Email notifications need to be configured and enabled in order to automatically generate passwords.',
   autogeneratepassword: 'Automatically generate password',
   validationEmail: 'You must provide a valid email address',
+  sortCreated: 'Creation Date',
+  sortUpdated: 'Last Updated',
+  sortUsername: 'Username',
```
Comment (sct) [inline src/components/UserList/index.tsx:67]: Should be "Display Name" and we should probably rename the table header as well.

Comment (TheCatLady) [reply src/components/UserList/index.tsx:67]: I thought that too, but I figured we should be consistent with the table header.  Changing this now :)

**Review by sct — APPROVED** [review]

**Pull request merged**